### ISSUE TITLE
Request: List of compatible audio players

### ISSUE THREAD

**Issue by thelung15** [mariusor/mpris-scrobbler#20]
Hi, sorry for making this an issue but I honestly don't know what music players are compatible with this program.

I tried Rhythmbox 3.4.1 with the "MPRIS D-Bus Interface" plugin set to **ON** (which is part of the rhythmbox-plugins ([3.4.1-2ubuntu5](https://packages.ubuntu.com/artful/rhythmbox-plugins)) Ubuntu "Artful" 17.10 package but it didn't successfully scrobble to my ListenBrainz account.

Maybe I did something wrong after I entered the ListenBrainz token because I got an error, but then I typed the same command again and got no error so I suppose there was no error in the end.

Can the community please help and list which gui music programs work for you(like Rhythmbox, Clementine etc.) and what versions of them along with potential plugins and their versions?

Thank you for having added support for ListenBrainz!

**Comment by mariusor** [comment]
Hi, I haven't really tested a lot of players. And theoretically any one should work. However the MPRIS specification is not very clear on which fields change trigger the event mpris-scrobbler listens for. 

The ones I know to be working are: Spotify, Audacious and if I recall correctly Clementine.

You can try to compile a debug version of the scrobbler and run it with the `-vvv` switch. Paste the output into a pastebin or a gist file and link it here. I can take a look to see what might be the issue.

**Comment by mariusor** [comment]
OK, I have installed Rhythmbox and with an already existing listenbrainz set-up it worked. 

To see if the signon was successful, check if the `grep -A2 listenbrainz ~/.local/share/mpris-scrobbler/credentials` shows the correct token (be careful not to paste it here, though :P).

To retry the sign-on, please call the `mpris-scrobbler-signon token listenbrainz -vvv` and paste the output here - I don't recall if any sensitive information is shown, so please take care of censoring the token if appears in the output.

**Comment by thelung15** [comment]
I typed `mpris-scrobbler-signon token listenbrainz -vvv` (and that includes the token I got from https://listenbrainz.org/profile/)(There's also a button on my listenbrainz.org profile page to reset my token, should I do that?) in my terminal and I got

> DEBUG   main::loading_credentials: ok
> DEBUG   saving::credentials[1]: /home/myuser/.config/mpris-scrobbler/credentials
> TRACING mem::freeing_configuration(1)

Then I typed `grep -A2 listenbrainz ~/.config/mpris-scrobbler/credentials` in my terminal and it returned

> [listenbrainz]
> enabled = false

Oh yeah, there seems to be some kind of problem I assume if it says **enabled = false**.

**Comment by mariusor** [comment]
Maybe the readme is nor clear enough. You shouldn't pass the token in the command line, the invocation of the command should ask for the token. At that step you can paste it
    
    $ mpris-scrobbler-signon token listenbrainz -vvv
    DEBUG   main::loading_credentials: ok
     INFO    signon::getting_token: listenbrainz.org
     Token for listenbrainz.org:

**Comment by thelung15** [comment]
Disclaimer: Before I also used this command `mpris-scrobbler-signon enable listenbrainz -vvv`. I don't know if it messed up stuff and if I should somehow reset mpris-scrobbler but anyway it returned

> DEBUG   main::loading_credentials: ok
> INFO    signon::enable: listenbrainz.org
> DEBUG   saving::credentials[1]: /home/myuser/.config/mpris-scrobbler/credentials
> DEBUG   signon::daemon_reload: unable to find PID file
> TRACING mem::freeing_configuration(1)
> TRACING mem::freeing_credentials(0x55d9d311a600): listenbrainz.org

Thank you. The command you posted I typed exactly 'as is'(thanks to your clarification) `mpris-scrobbler-signon token listenbrainz -vvv` and then it asked me for the token. I pasted it in and later I used the command `grep -A2 listenbrainz ~/.config/mpris-scrobbler/credentials` and it returned:

> [listenbrainz]
> enabled = true
> token = 99dc1923-23f2-498f-9400-dcdbd5eef954

**Comment by mariusor** [comment]
I censored your token above :D 

And it looks OK otherwise, does the scrobbler send tracks to listenbrainz in this configuration?

**Comment by thelung15** [comment]
Disclaimer again: After I realized my mistake of posting my token I reset it and now it is something different. I typed `mpris-scrobbler-signon token listenbrainz -vvv` and entered the new token and used `grep -A2 listenbrainz ~/.config/mpris-scrobbler/credentials` and this time when quoting what I got as a returned result I'll censor the token but I've verified it's the correct one:

> [listenbrainz]
> enabled = true
> token = the new token, which I've double-checked to make sure it's the correct one

I'll also restart Rhythmbox now and play some songs and if it works I'll edit this post at the top(and if it doesn't I'll still edit it at the top)

**Comment by mariusor** [comment]
Sure. If it works, you can close the issue directly.

Also, if you have a suggestion about how the readme could be more clear about the steps to log-in to listenbrainz, feel free to create a PR and I'll take a look.

**Comment by thelung15** [comment]
I just played Josh Woodward - Swansong (full length: 4m 20s) and it did not scrobble to my user page https://listenbrainz.org/user/internetofdata

I don't have a suggestion for the readme. I think it is good as it is. What I'd do if I make it work or if @jflory7 makes it work on an own setup with the same versions of the programs I've used and it works I think that's enough to add it to a public list of apps that support(more or less the listenbrainz API) over at 

https://community.metabrainz.org/t/list-of-native-listenbrainz-api-apps-and-in-development-apps/342938

**Comment by mariusor** [comment]
OK, damn.

Can you please run the scrobbler with verbose output `mpris-scrobbler -vvv` and then play some songs?

**Comment by thelung15** [comment]
Yeah, wow. Cool! Thanks for your support.

> TRACING mem::scrobble::freeing_title(0x55e7fe05fd50): Border Blaster
> TRACING mem::scrobble::freeing_album(0x55e7fe05ff60): Breadcrumbs
> TRACING mem::scrobble:freeing_artist(0x55e7fe060170): Josh Woodward
> TRACING mem::freeing_scrobble(0x55e7fe057b60)
It definitely got the song title Border Blaster right(btw it's also one of my favorite songs)

**Comment by mariusor** [comment]
Please post the whole scrobbler session, at least 4 minutes of play time. You can use a gist or a paste service if the amount of text is to large for github.

**Comment by thelung15** [comment]
https://gist.github.com/thelung15/82ff927f448fdd79ba9cb33aca12b196

**Comment by thelung15** [comment]
Somehow it worked: can be seen here https://listenbrainz.org/user/internetofdata I see 2 successful "listens"(uncommon word for "scrobble" that the ListenBrainz project uses to avoid trademark infringement in relation to last.fm) though do I always need to use the `mpris-scrobbler -vvv` for this to work?

**Comment by mariusor** [comment]
I suspect you're not patient enough to see your track. 

According to the log it would have sent it after 142 seconds of play (which is the half of the song): https://gist.github.com/thelung15/82ff927f448fdd79ba9cb33aca12b196#file-listenbrainz-txt-L234

According to the API documentation, a song should be marked as listened only after 4 minutes or half of it played, which ever comes first.

**Comment by mariusor** [comment]
And no, -vvv is used just to see the output, it should have no impact on how it the scrobbler works.

**Comment by thelung15** [comment]
Should I open a new issue or do you know if mpris-scrobbler fully supports the ListenBrainz API? This is what a user's page at ListenBrainz looks like when a program has full support for the ListenBrainz API: https://listenbrainz.org/user/marc2k3. As can be seen artist name is clickable and track name as well. The songs I scrobbled through Rhythmbox were .ogg songs that I had entered with MusicBrainz Picard and all the MBIDs are present inside all these files but maybe this is an issue with Rhythmbox that can't see the MBIDs inside my files. Do you know if mpris-scrobbler can see them?

**Comment by mariusor** [comment]
Unfortunately the MPRIS API doesn't support the custom music brainz tags that allow linking artists and songs. Feel free to open an issue, but I'm not sure it will ever get closed.

**Comment by thelung15** [comment]
Wow. Thank you for that input!

**Comment by jflory7** [comment]
> Unfortunately the MPRIS API doesn't support the custom music brainz tags that allow linking artists and songs. Feel free to open an issue, but I'm not sure it will ever get closed.

@mariusor I'm curious, why is this? Is the support in ListenBrainz not there? I wonder if we should file a new feature request in ListenBrainz for this if not.

**Comment by mariusor** [comment]
The definitions of the metadata that can be loaded with MPRIS (the website is down at the moment, but I'll add a link later) don't include the hashes set by musicbrainz. It has nothing to do with listenbrainz.

**Comment by thelung15** [comment]
@mariusor I assume it is the Freedesktop.org site? I'm having trouble reaching it as well

**Comment by mariusor** [comment]
Yes.

**Comment by thelung15** [comment]
Before this is closed I think you told me that the command `mpris-scrobbler -vvv` would not help my songs get "scrobbled/listened" but I stopped the command and I played Josh Woodward's "Cherry" and it did not show up in now playing on my page. After I started the `mpris-scrobbler -vvv` again then it showed up correctly. The now playing works perfectly and responds **immediately** so regardless of how patient or impatient anybody is they will get the now playing immediately after running the command `mpris-scrobbler -vvv`. That's my conclusion. I'm sorry if this makes 0 sense.

**Comment by mariusor** [comment]
How do you run it otherwise?

**Comment by thelung15** [comment]
I didn't know. I assumed it always ran in the background. Oh, I just think I learnt something new from your short answer. So one always has to type ./mpris-scrobbler, ok I see now. Well thanks for this educational session :) and I'm still considering donating something, maybe if you don't think monetary donations are worthwhile maybe I'll add issues to other projects which are related to MPRIS as thanks or if you got any other ideas?

**Comment by mariusor** [comment]
I don't have a way to receive donations, sorry. If you want to donate, please support libre.fm or listenbrainz directly, they have actual costs for their infrastructure.

In the documentation it's mentioned how to keep mpris-scrobbler running using systemd.

I will close the issue now, as I feel we've solved it successfully. :)

**Comment by mariusor** [comment]
Guys, see #23 and #24 for adding music brainz tags to the listenbrainz submitted tracks.
